raincloudy/controller.py

```diff
@@ -141,8 +114,9 @@ def id(self):
     def name(self):
         """Return controller name."""
         return \
-            find_controller_or_faucet_name(self._parent.html['home'],
-                                           'controller')
+            find_controller_or_faucet_name(self.home,
+                                           'controller',
+                                           self._index)
```

Comment:
adding the index to the instance is a bit weird. but it lets the instance know where it is on the page so it can self-update later